messaging/src/main/java/ids/messaging/protocol/http/IdsHttpService.java

```diff
@@ -399,7 +412,15 @@ private Response sendRequest(final Request request,
             log.info("Sending request to {} ...", request.url());
         }
 
-        final var response = client.newCall(request).execute();
+        var response = client.newCall(request).execute();
+        final var responseBody = response.body();
+
+        if (Boolean.TRUE.equals(logResponses) && responseBody != null) {
+            final var bodyString = responseBody.string();
+            log.info("Incoming response body: {}", bodyString);
+            final var body = ResponseBody.create(bodyString, responseBody.contentType());
```

Comment:
Is this line wrapping again the response in a new response object?
Accessors of the body I found so far, are:
src/main/java/ids/messaging/protocol/http/IdsHttpService.java:125
src/test/java/ids/messaging/protocol/http/IdsHttpServiceTest.java:131

Reply:
Yes, because the `responseBody` itself can only be consumed once, which is done by `.string()`. You have to create a kind of copy after consuming it to resume.

Reply:
Good workaround!